src/DynamoCore/Engine/NodeToCode/NodeToCode.cs

```diff
@@ -1210,6 +1212,26 @@ public static void ReplaceWithShortestQualifiedName(ClassTable classTable, IEnum
             }
             #endregion
 
+            #region Step 5 Escape StringInputNodes
+            foreach ((NodeModel nodeModel, IEnumerable<AssociativeNode> astNodes) in allAstNodes)
+            {
+                if (nodeModel.NodeType != "StringInputNode")
```

Comment:
I don't believe it's possible to reference `CoreNodeModels.Input.StringInput` from the `DynamoCore` project as `DynamoCore` is itself a reference of `CoreNodeModels` so it would lead to a circular dependency.

Is there a different way to identify the StringInput node?

Reply:
Sorry, I missed that. In that case, I would suggest you use the Visitor pattern we have several classes for. I'd suggest you add a new class like the following:
```
internal class StringReplacer : AstReplacer
    {
        public override AssociativeNode VisitStringNode(StringNode node)
        {
            var escapedString = node.Value.Replace(@"\", @"\\")
                .Replace("\"", "\\\"");
            node.Value = escapedString;
            return node;
        }
    }
```
Then call this in place of your code above like this:
```
#region Step 5 Escape StringInputNodes

            foreach ((NodeModel nodeModel, IEnumerable<AssociativeNode> astNodes) in allAstNodes)
            {
                var stringReplacer = new StringReplacer();
                stringReplacer.VisitNodeList(astNodes.ToList());
            }

            #endregion
```

Reply:
You can make `StringReplacer` a private nested class within `NodeToCodeCompiler` since it's most likely going to be used only in the node to code context.

Reply:
Thanks @aparajit-pratap 

I tried implementing the Visitor method but it led to the following tests failing:

![image](https://user-images.githubusercontent.com/193290/101023773-272d8700-356b-11eb-966a-d1832bb74d80.png)

This seems to occur as nodes like the DirectoryPath also trigger the `StringReplacer.VisitStringNode` method which leads to them becoming escaped twice. For the path `C:\` you would expect `C:\\` But instead you get `C:\\\\`

It seems the Directory Path is already escaped in the `BuildAst` method: https://github.com/DynamoDS/Dynamo/blob/b3b5c097533e1e43020f6a2fc8f82aa185e33925/src/Libraries/CoreNodeModels/Input/FileSystem.cs#L49-L55

As this looks like a cleaner solution I've adopted the same approach for the `StringInput` node in commit c39adef and reverted the earlier changes I made to `NodeToCode.cs`. Please let me know your thoughts on this?